source/contracts/libraries/Extractable.sol

```diff
@@ -0,0 +1,34 @@
+pragma solidity 0.4.17;
+
+import 'libraries/token/ERC20Basic.sol';
+import 'Controlled.sol';
+
+
+contract Extractable is Controlled {
+    // Send accidentally received ETH to the destination
+    function extractETH(address _destination) public onlyControllerCaller returns(bool) {
+        require(mayExtract(ERC20Basic(0)));
+        require(_destination.call.value(this.balance)());
+        return true;
+    }
+
+    // Send accidentally received tokens to the destination
+    function extractTokens(address _destination, ERC20Basic _token) public onlyControllerCaller returns (bool) {
+        require(mayExtract(_token));
+        uint256 _balance = _token.balanceOf(this);
+        _token.transfer(_destination, _balance);
+        return true;
+    }
+
+    function mayExtract(ERC20Basic _token) private returns (bool) {
+        address[] memory _protectedTokens = getProtectedTokens();
+        for (uint8 i = 0; i < _protectedTokens.length; i++) {
+            if (_protectedTokens[i] == address(_token)) {
+                return false;
+            }
+        }
```

Comment:
`.contains` seems like a good candidate for the `*Array` libraries.  This would allow us to unit test the `.contains` code separately from the business logic around getting protected tokens.  I know this is a minor gain, but I think generally a good abstraction.

Reply:
Caveat: That would mean that all contracts that derive from this will need to compile in `AddressArray`, which also contains slice.  Due to the contract size limits, I wonder if we would be better off having a separate contract for each of the array methods?  Maybe the same for the safe math libraries?

```
contract Apple {
    using AddressArrayContains for address[];
    using Uint256SafeMathAdd for uint256;
    using Uint256SafeMathSub for uint256;

    function banana(address[] array, uint256 number) returns (uint256) {
        return array.sum().add(5).sub(7);
    }
}
````

The above pattern would allow us to shrink down some of the contracts that use a subset of SafeMath (every contract that uses SafeMath uses a subset of it) and it would allow us to be a bit more liberal with what we convert to libraries.

Note: We can put multiple contracts into the same file, so we could still have `SafeMathUint256.sol`, it would just have multiple contracts in it (one for each function).

Reply:
Noted in Slack too but for posterity: punting this till later and tracking as a CH issue